edge/key.go

```diff
@@ -120,15 +119,15 @@ func encodeKey(edgeKey *edgeKey) string {
 	return encodedKey
 }
 
-func RetrieveEdgeKey(edgeKey string, clusterService agent.ClusterService) (string, error) {
+func RetrieveEdgeKey(dataPath string, edgeKey string, clusterService agent.ClusterService) (string, error) {
```

Comment:
Why does the `RetrieveEdgeKey` function needs to know about the path and the `SetKey` doesn't ? Maybe make it consistent and have `(manager *Manager) RetrieveEdgeKey` instead.